src/lib/Syntax.hs

```diff
@@ -156,13 +157,15 @@ data Atom (n::S) =
    deriving (Show, Generic)
 
 data Expr n =
-   App (Atom n) (Atom n)
+   App (Atom n) [Atom n]
  | Case (Atom n) [Alt n] (Type n) (EffectRow n)
- | Atom (Atom n)
  | Op  (Op  n)
  | Hof (Hof n)
    deriving (Show, Generic)
 
+pattern Atom :: Atom n -> Expr n
+pattern Atom x = App x []
```

Comment:
I have to say this is a little sketchy 😬

Reply:
Yeah, you're right. I was trying to save a case but we end up treating them separately in several places anyway, and the dependence on case order becomes a footgun. Let's keep `Atom` as a separate case and make `App` n-ary for n>0.

I merged this one already but I'll make a follow-up PR.